### PR TITLE
Added 64bit VertexAttributeValues and Conversions

### PR DESCRIPTION
**Pull request by TheBiochemic**
# Objective

The Goal is to add the missing f64 VertexAttributeValues Types, that exist in wgpu, but were not present in bevy.

## Solution

Extended the VertexAttributeValues enum with the Float64, Float64x2, Float64x3 and Float64x4 Variants,
added some Tests for these new Types and also checked, if the new Vertex Attributes can be added into a Mesh.
Also Added a performance Note, for those who want to explore this area further.

---

## Changelog

* Added VertexAttributeValues::Float64 enum variant
* Added VertexAttributeValues::Float64x2 enum variant
* Added VertexAttributeValues::Float64x3 enum variant
* Added VertexAttributeValues::Float64x4 enum variant
* Added Appropriate Conversion Methods for Arrays of f64
* Added Tests for checking, if adding f64 Array is possible into a mesh
* Added Tests to cover f64 conversions

## Migration Guide

- Matches on the enum VertexAttributeValues will break, fixable by just adding the new variants in the match.
- if using `mesh.insert_attribute(...)` make sure, the inbound Vec is of the correct floating point type. It sometimes can happen, if not restricted by type anywhere, that a f64 type might be used automatically. The error message of not doing that will contain something along the lines of (analog also other variants):

> Failed to insert attribute. Invalid attribute format for Vertex_Uv. Given format is Float64x2 but expected Float32x2

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
This is technically breaking, since it adds variants to a public enum that could have been handled exhaustively via match. Not a problem here, but figured I'd let you know.

File: crates/bevy_render/src/mesh/mesh/mod.rs [hunk 1]
```diff
@@ -679,6 +683,12 @@ impl VertexFormatSize for wgpu::VertexFormat {
 
 /// Contains an array where each entry describes a property of a single vertex.
 /// Matches the [`VertexFormats`](VertexFormat).
+///
+/// Be aware that the [`Float64`](VertexAttributeValues::Float64),
+/// [`Float64x2`](VertexAttributeValues::Float64x2), [`Float64x3`](VertexAttributeValues::Float64x3) and
+/// [`Float64x4`](VertexAttributeValues::Float64x4) need to be used with Caution, because they might cause
+/// Performance Problems due to [f64](std::primitive::f64) being between [considerably](https://groups.google.com/g/amgcl/c/Bz_X2B0UBWE)
```
Comment (jpsikstus) [inline crates/bevy_render/src/mesh/mesh/mod.rs:690]: Looks like the word "between" shouldn't be here.

Comment (jpsikstus) [reply crates/bevy_render/src/mesh/mesh/mod.rs:690]: "Caution" should probably be lowercase.

Comment (TheBiochemic) [reply crates/bevy_render/src/mesh/mesh/mod.rs:690]: Will fix it real quick, thank you :)

**Comment by TheBiochemic** [comment]
@alice-i-cecile you're right. I'm currently running the examples, and apparently because i added the conversions it might be actually breaking, because it seems to mess with the default shapes when the type is not explicitly defined. It then assumes the biggest possible floating point type, which is f64, whenever possible. I'm currently adjusting it and will change the description later too, thanks for pointing it out!

**Comment by fintelia** [comment]
Is is probably worth mentioning in the documentation somewhat that using 64-bit vertex attributes requires a wgpu feature to be enabled and isn't supported on all hardware. In particular, my understanding is that Metal doesn't expose any support for 64-bit floats

**Comment by TheBiochemic** [comment]
@fintelia 

Yeah i have read about that, and i'm not sure how to approach this exactly yet. What would be a good strategy to tackle this?
Does a feature flag make sense?

**Comment by alice-i-cecile** [comment]
If that's the case, we should probably gate these enum variants by that feature.

**Comment by fintelia** [comment]
My understanding is that wgpu are controlled at runtime by the [`WgpuFeatures` resource](https://docs.rs/bevy/latest/bevy/render/render_resource/struct.WgpuFeatures.html#). In this particular case, the relevant feature would be [`VERTEX_ATTRIBUTE_64BIT`](https://docs.rs/bevy/latest/bevy/render/render_resource/struct.WgpuFeatures.html#associatedconstant.VERTEX_ATTRIBUTE_64BIT). This is a different concept than cargo features.

**Comment by alice-i-cecile** [comment]
Ah I see. That's fine then, just mention it prominently in the docs.

**Comment by alice-i-cecile** [comment]
Docs look good, thanks! I'll wait for more rendering expertise on this one.

**Comment by JMS55** [comment]
Getting to this as part of a triage of really old PRs. I'm opposed to adding this unless there's a very compelling use case, personally.

**Pull request closed without merging**